### ISSUE TITLE
VIT Example Performance Issue

### ISSUE THREAD

**Issue by sunweilunformatech** [keras-team/keras-io#901]
Hi Keras team,

I came across the ViT example [here](https://keras.io/examples/vision/image_classification_with_vision_transformer/). However, I found that the same performance can be achieved by simply replacing the multi-head attention layer with a simple dense layer. So I believe the example does not demonstrate the advantage of transformer based architectures. Any ideas on why this happens?

Specifically, I simply replace the create create_vit_classifier() function in the demo notebook with the following:
```
def create_vit_classifier():
    inputs = layers.Input(shape=input_shape)
    # Augment data.
    augmented = data_augmentation(inputs)
    # Create patches.
    patches = Patches(patch_size)(augmented)
    # Encode patches.
    encoded_patches = PatchEncoder(num_patches, projection_dim)(patches)

    for _ in range(transformer_layers):
        x1 = encoded_patches
        x2 = mlp(x1, hidden_units=[x1.shape[-1]], dropout_rate=0.1)
        x2 = layers.Add()([x1, x2])
        x3 = mlp(x2, hidden_units=transformer_units, dropout_rate=0.1)
        encoded_patches = layers.Add()([x1, x3])
                              
    # Create a [batch_size, projection_dim] tensor.
    representation = layers.LayerNormalization(epsilon=1e-6)(encoded_patches)
    representation = layers.Flatten()(representation)
    representation = layers.Dropout(0.5)(representation)
    # Add MLP.
    features = mlp(representation, hidden_units=mlp_head_units, dropout_rate=0.5)
    # Classify outputs.
    logits = layers.Dense(num_classes)(features)
    # Create the Keras model.
    model = keras.Model(inputs=inputs, outputs=logits)
    return model
 ```
The results after training 100 epochs is:
Test accuracy: 54.93%
Test top 5 accuracy: 81.93%

Which has no material difference with the transformer model:
Test accuracy: 55.12%
Test top 5 accuracy: 82.0%

**Comment by sunweilunformatech** [comment]
Also note that replacing MHA with dense layer makes the training 2x faster.

**Comment by sachinprasadhs** [comment]
@sunweilunformatech , Could  you please create a PR with the changes you suggested. Thanks
